File: GuardianX-main/backend/app/schemas/scan.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.core.scan_profile import ScanProfile
from app.core.scan_status import ScanStatus


class ScanCreate(BaseModel):
    asset_id: int
    scan_profile: ScanProfile = ScanProfile.STANDARD


class ScanResponse(BaseModel):
    id: int
    asset_id: int
    asset_name: str | None = None
    status: ScanStatus
    scanner: str
    scan_profile: ScanProfile = ScanProfile.STANDARD
    started_at: datetime | None
    finished_at: datetime | None
    created_at: datetime
    finding_count: int = 0

    model_config = ConfigDict(
        from_attributes=True,
    )


class ScanResultResponse(BaseModel):
    id: int
    port: int
    protocol: str
    state: str
    service: str | None
    product: str | None
    version: str | None
    is_ssl: bool

    model_config = ConfigDict(
        from_attributes=True,
    )


class ExecutorStatus(BaseModel):
    max_workers: int
    queued: int
    running: int
    idle_workers: int
    closed: bool


class ScanOperationsResponse(BaseModel):
    executor: ExecutorStatus
    counts: dict[str, int]
    total: int

File: GuardianX-main/guardianx-frontend/src/types/scan.ts
export type ScanStatus =
  | "PENDING"
  | "RUNNING"
  | "COMPLETED"
  | "FAILED"
  | "CANCELLED";

export type ScanProfile = "standard" | "full";

export interface Scan {
  id: number;
  asset_id: number;
  asset_name: string | null;
  status: ScanStatus;
  scanner: string;
  scan_profile: ScanProfile;
  started_at: string | null;
  finished_at: string | null;
  created_at: string;
  finding_count: number;
}

export interface ScanResult {
  id: number;
  port: number;
  protocol: string;
  state: string;
  service: string | null;
  product: string | null;
  version: string | null;
  is_ssl: boolean;
}

export interface CreateScanDto {
  asset_id: number;
  scan_profile?: ScanProfile;
}

export interface ExecutorStatus {
  max_workers: number;
  queued: number;
  running: number;
  idle_workers: number;
  closed: boolean;
}

export interface ScanOperations {
  executor: ExecutorStatus;
  counts: Record<string, number>;
  total: number;
}
